### ISSUE TITLE
bug(LiveAnnouncer): LiveAnnouncer does not announce subsequent calls to the same text

### ISSUE THREAD

**Issue by DmitryEfimenko** [angular/components#21507]
#### Reproduction

Use StackBlitz to reproduce your issue: 
- Components: https://stackblitz.com/edit/angular-announcer-bug?file=src/app/app.component.html

Steps to reproduce:
1.  Open StackBlitz link from above
2. Enable Narrator (on Windows press Windows key + Ctrl + Enter)
3. Click on "Hello world" button multiple times
4. Observe that Narrator announces text only on the first click

Note that if text to announce changes, the announcer works as expected. You can reproduce that by pressing on "Announce Different" button multiple times.
 
#### Expected Behavior
Narrator should announce text on each click

#### Actual Behavior
Narrator  announces text only on the first click if the text being announced didn't change

#### Environment

- Angular:
- CDK/Material: 11.0.3
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by jelbourn** [comment]
@DmitryEfimenko does this behavior happen if you `clear` before making the announcement (likely requiring a small `setTimeout`). My suspicion is that the `aria-live` element doesn't announce anything because it's content didn't actually change, which could be considered an expected behavior. 

cc @stevenyxu since this might relate to the concept of queueing live announcer messages, which might provide a way to deal with this.

**Comment by DmitryEfimenko** [comment]
It seems that this use-case was accounted for according to the comments in the [source code](https://github.com/angular/components/blob/7beb95219ddc9a16cd28ee2524bf1594e04c1c58/src/cdk/a11y/live-announcer/live-announcer.ts#L112-L127):
```ts
// This 100ms timeout is necessary for some browser + screen-reader combinations:
// - Both JAWS and NVDA over IE11 will not announce anything without a non-zero timeout.
// - With Chrome and IE11 with NVDA or JAWS, a repeated (identical) message won't be read a
//   second time without clearing and then using a non-zero delay.
// (using JAWS 17 at time of this writing).
```
According to the above, the expected behavior is that repeated identical messages should be announced.

Also, looking through the code it looks like it already clears the text and sets timeout to handle exactly this use-case, but for some reason this does not work.
I'd say this is a regression.

**Comment by jelbourn** [comment]
It's unlikely a regression because this code hasn't changed since inception. Does the behavior occur in screen-readers other than Narrator? We've never explicitly supported that one (just JAWS and NVDA on Windows), so I'm wondering if it's a reader-specific behavior.

**Comment by DmitryEfimenko** [comment]
I haven't test with JAWS since it's a payed version of software. However, I installed NVDA and it looks like it works as expected with it.
I'd think the Narrator should be supported as well since it's a default screen reader that comes with Windows.

**Comment by mkhanal** [comment]
The issue also occurs with both Edge and Chrome on Mac with voiceover.

**Comment by amysorto** [comment]
As per https://github.com/angular/components#browser-and-screen-reader-support, we don't support Windows with Narrator or Mac with Edge. I tested on Chrome with Voiceover and it seems to be working there.
